support/build-debian.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/sh
+# Copyright 2024 Northern.tech AS
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+DEB_ARCH="${1}"
+export GOOS="linux"
+case "${DEB_ARCH}" in
+    "armhf")
+        export GOARCH="arm"
+        export GOARM="6"
+        ;;
+    "arm64")
+        export GOARCH="arm64"
+        ;;
+    "amd64")
+        export GOARCH="amd64"
+        ;;
+    *)
+        echo "Usage: $0 <DEBARCH>" 1>&2
+        exit 1
+        ;;
+esac
+VERSION="$(git describe --tags --dirty)"
+TMPDIR="$(mktemp --tmpdir -d "nt-connect-XXXXX")"
+trap 'rm -rf $TMPDIR' EXIT
+export DESTDIR="${TMPDIR}/nt-connect-${VERSION#v}_${GOOS}_${DEB_ARCH}"
+mkdir -p "${DESTDIR}/DEBIAN"
+make install
+cat > "${DESTDIR}/DEBIAN/control" <<EOF
```

Comment:
Having this in the same source tree than the production code, and on top of it generate this files on the fly is a bad practice.

The main objection is that if you have an already tagged production software (say 1.0.0) then you won't be able to modify the recipe or scripts without re-tagging.

All build systems that I know about have two versions for this. Debian specifically has the project version (1.0.0) and then an append of the Debian version (`-1`, `-2`, etc) for this reason.

I recommend we do this more streamlined, either on a separate branch or on a separate repo.

Reply:
>The main objection is that if you have an already tagged production software (say 1.0.0) then you won't be able to modify the recipe or scripts without re-tagging.

In general I agree, but on the other hand I don't see why we would want to modify the "recipe" for a Debian package where we are also maintaining the source code. This pattern makes sense if the package maintainer is not maintaining the source code, but the reason I went this way was just to keep it as simple as possible.

> All build systems that I know about have two versions for this. Debian specifically has the project version (1.0.0) and then an append of the Debian version (-1, -2, etc) for this reason.

We are not relying on any external dependencies, we are building a static binary and distributing it with some application-specific configuration files (and a systemd file). So I don't see any reason to build for the different distro flavors.

Reply:
> why we would want to modify the "recipe" for a Debian package

We had many cases in Mender packages where a file got installed in the wrong place, or something needed to be amended in the package.

> (...) the reason I went this way was just to keep it as simple as possible.

In my experience this argument blows up eventually, it is a matter of time. We tend to re-use what we have so it is better to build things to last. For example, have you thought about bugfix releases? Are we going to support more than one branch at some point? Then fixes on the build scripts, dependencies or the "recipe" will need porting and will mess up the tags.